src/main/java/controllers/impl/UserControllerImpl.java

```diff
@@ -0,0 +1,54 @@
+package controllers.impl;
+
+import controllers.UserController;
+import dto.UserResponseDto;
+import java.util.List;
+import java.util.stream.Collectors;
+import models.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import service.UserMapper;
+import service.UserService;
+
+@RestController
+@RequestMapping("/user")
+public class UserControllerImpl implements UserController {
+    private final UserService userService;
```

Comment:
UserMapper should also be a dependency here - don't make the methods static there